cni/pkg/cmd/root.go

```diff
@@ -85,14 +85,13 @@ var rootCmd = &cobra.Command{
 
 		repair.StartRepair(ctx, &cfg.RepairConfig)
 
-		if err = installer.Run(ctx); err != nil {
```

Comment:
`if err := installer.Run(ctx); err != nil {`

does that fix instead?

Reply:
That's the original code which fails. I believe the linter was looking at Run and sees that it would never return an err that was nil, so decided there is no reason for the nil check of the returned value.

Reply:
the `:=` vs `=` was the difference here. Shadowing here seems concerning, I feel like I am not understanding something

Reply:
Ahh, missed the `:`. If I try that I get:
```
cni/pkg/cmd/root.go:92:5: ineffectual assignment to err (ineffassign)
				err = nil
```
I believe the main issue is Run uses a passed in `err`:
```
func (in *Installer) Run(ctx context.Context) (err error) {
```

Reply:
I did restructure the Run to just have error and then used := for the err assignments in the function nd changed the return's to be return err, and that also passes the lint check.